cpu/nrf52/radio/nrf802154/nrf802154_radio.c

```diff
@@ -124,6 +124,11 @@ static bool _l2filter(uint8_t *mhr)
 
     int addr_len = ieee802154_get_dst(mhr, dst_addr, &dst_pan);
 
+    if ((mhr[0] & IEEE802154_FCF_TYPE_MASK) == IEEE802154_FCF_TYPE_BEACON) {
```

Comment:
In earlier days (before 802.15.4 radio HAL) GNRC got stuck receiving beacon frames from devices that did not filter those in hardware. Clearly, you need to receive beacons to enable a proper MAC. I just want to make sure we don't break the "standard" CSMA sender with this.

Reply:
In case this is a problem, I think these stuff should be fixed in an upper layer (e.g SubMAC). Otherwise we will end up with too many corner cases at driver level, because _some_ modules need it and _some_ not

Reply:
Have you tested on this PR what happens in a standard CSMA case when a beacon is received?

I agree that the driver is not the best place for it (actually for any of the common 802.15.4 filter routines) but in the first place we should not introduce a feature that breaks the (currently) most common use case.

Reply:
GNRC should just discard the frame if it finds it to not be a valid 6LoWPAN frame, right @miri64?

Reply:
> GNRC should just discard the frame if it finds it to not be a valid 6LoWPAN frame, right @miri64?

No, `gnrc_sixlowpan` will do that, yes, but GNRC will just eat up anything you give it ;-). If there is a Zigbee handler it will also handle Zigbee packets (which are not valid 6LoWPAN frames IIRC).